From 4e1d6d4b5ca42db24433a1596094eb807237c7c6 Mon Sep 17 00:00:00 2001
From: "N, Shyjumon" <shyjumon.n@intel.com>
Date: Wed, 12 Aug 2020 01:16:32 +0530
Subject: [PATCH] Merging the applied build changes

patches:
	1. REVERTME-Disable-kati-checks-and-compiler-warning
    	   I735f0c2c6fab153e9991e27fb08677723764afe5
	    Suppress compiler warnings
	    -Wno-tautological-bitwise-compare
	    -Werror=pragma-pack
	    -Werror=pragma-pack-suspicious-include
	    Disable below kati checks
	    --werror_overriding_commands
	    --werror_real_to_phony
	    --werror_phony_looks_real
	    --werror_writable
	2. Disable-path-restriction
	   I612f6e8c44a80683311dc8f1b076fe7fa8d40a67

Change-Id: I08b543dca54b1ed30ee2b84a6ea7291cc1ad4994
Tracked-On: None
Signed-off-by: ahs <amrita.h.s@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Signed-off-by: N, Shyjumon <shyjumon.n@intel.com>
---
 cc/config/clang.go  | 1 +
 cc/config/global.go | 2 --
 ui/build/kati.go    | 5 +----
 ui/build/path.go    | 2 +-
 4 files changed, 3 insertions(+), 7 deletions(-)

diff --git a/cc/config/clang.go b/cc/config/clang.go
index 2e0b24183..4655466d8 100644
--- a/cc/config/clang.go
+++ b/cc/config/clang.go
@@ -172,6 +172,7 @@ func init() {
 		"-Wno-int-in-bool-context",          // http://b/148287349
 		"-Wno-sizeof-array-div",             // http://b/148815709
 		"-Wno-tautological-overlap-compare", // http://b/148815696
+		"-Wno-tautological-bitwise-compare",
 		// New warnings to be fixed after clang-r383902.
 		"-Wno-deprecated-copy",                      // http://b/153746672
 		"-Wno-range-loop-construct",                 // http://b/153747076
diff --git a/cc/config/global.go b/cc/config/global.go
index fce03061a..2bb6cbb45 100644
--- a/cc/config/global.go
+++ b/cc/config/global.go
@@ -49,8 +49,6 @@ var (
 		"-fno-strict-aliasing",

 		"-Werror=date-time",
-		"-Werror=pragma-pack",
-		"-Werror=pragma-pack-suspicious-include",
 	}

 	commonGlobalConlyflags = []string{}
diff --git a/ui/build/kati.go b/ui/build/kati.go
index a845c5be2..7b87e7258 100644
--- a/ui/build/kati.go
+++ b/ui/build/kati.go
@@ -78,9 +78,6 @@ func runKati(ctx Context, config Config, extraSuffix string, args []string, envF
 		"--werror_suffix_rules",
 		"--warn_real_to_phony",
 		"--warn_phony_looks_real",
-		"--werror_real_to_phony",
-		"--werror_phony_looks_real",
-		"--werror_writable",
 		"--top_level_phony",
 		"--kati_stats",
 	}, args...)
@@ -139,7 +136,7 @@ func runKatiBuild(ctx Context, config Config) {
 	}

 	if !config.BuildBrokenDupRules() {
-		args = append(args, "--werror_overriding_commands")
+		//args = append(args, "--werror_overriding_commands")
 	}

 	args = append(args, config.KatiArgs()...)
diff --git a/ui/build/path.go b/ui/build/path.go
index 84e999a4b..b97c692ff 100644
--- a/ui/build/path.go
+++ b/ui/build/path.go
@@ -95,7 +95,7 @@ func SetupLitePath(ctx Context, config Config) {
 	prebuiltsPath, _ := filepath.Abs("prebuilts/build-tools/path/" + runtime.GOOS + "-x86")
 	myPath = prebuiltsPath + string(os.PathListSeparator) + myPath

-	config.Environment().Set("PATH", myPath)
+	//config.Environment().Set("PATH", myPath)
 	config.pathReplaced = true
 }

--
2.17.1

